### PR TITLE
Adding support for Prisma's datamodel syntax

### PR DESCRIPTION
**Pull request by rajinwonderland**
Adding support for [Prisma 2's](https://www.prisma.io/blog/announcing-prisma-2-zq1s745db8i5) new datamodel syntax for `.prisma` extensions.

> Not sure if I've got everything, so all feedback/suggestions are welcome! 😀

## Description
Adds support and grammar for [Prisma 2's](https://www.prisma.io/blog/announcing-prisma-2-zq1s745db8i5) new datamodel syntax language for `.prisma` files.

> Information about the new datamodel syntax can be found [on Prisma 2's docs](https://github.com/prisma/prisma2/blob/master/docs/data-modeling.md#example).

**Documentation References**
- [Prisma 2's Data Modeling docs](https://github.com/prisma/prisma2/blob/master/docs/data-modeling.md#example)
- [Prisma schema file usage docs](https://github.com/prisma/prisma2/blob/master/docs/prisma-schema-file.md)
 
## Checklist:

- [x] **I am adding a new language.**
  - [x] The extension of the new language is used in hundreds of repositories on GitHub.com.
    - Search results for each extension:
      
      -  https://github.com/search?q=extension%3Aprisma+NOT+nothack&type=Code
  - [x] I have included a real-world usage sample for all extensions added in this PR:
    - Sample source(s):
       - [schema.prisma](https://github.com/prisma/photonjs/blob/master/examples/typescript/graphql/prisma/schema.prisma)
       - [datasource-mongo.prisma](https://github.com/prisma/prisma2/blob/master/docs/prisma-schema-file.md)
       - [datasource-postgres.prisma](https://github.com/prisma/prisma2/blob/master/docs/prisma-schema-file.md)
       - [datasource-mysql.prisma](https://github.com/prisma/prisma2/blob/master/docs/prisma-schema-file.md)
       - [datasource-sqlite.prisma](https://github.com/prisma/prisma2/blob/master/docs/prisma-schema-file.md)
    - Sample license(s):
       - All are under `apache-2.0`
  
  - [x] I have included a syntax highlighting grammar: [github-lightshow](https://github-lightshow.herokuapp.com/?utf8=%E2%9C%93&scope=from-url&grammar_format=auto&grammar_url=https%3A%2F%2Fgithub.com%2Fprisma%2Fvscode-prisma%2Fblob%2Fmaster%2Fsyntaxes%2Fprisma.tmLanguage.json&grammar_text=&code_source=from-url&code_url=https%3A%2F%2Fraw.githubusercontent.com%2Fprisma%2Fphotonjs%2Fmaster%2Fexamples%2Ftypescript%2Fgraphql%2Fprisma%2Fschema.prisma&code=)
  - [ ] I have updated the heuristics to distinguish my language from others using the same extension.

> Not sure if the section below is applicable?
- [ ] **I am changing the color associated with a language**
  
  - [ ] I have obtained agreement from the wider language community on this color change.
    - [URL to public discussion]
    - [Optional: URL to official branding guidelines for the language]

### CODE REVIEW

File: lib/linguist/languages.yml [hunk 1]
```diff
@@ -41,33 +41,33 @@
   type: programming
   color: "#814CCC"
   extensions:
-  - ".bsl"
-  - ".os"
+    - ".bsl"
+    - ".os"
```
Comment (pchaigno) [inline lib/linguist/languages.yml]: Any idea what happened here?

File: .gitmodules [hunk 2]
```diff
@@ -965,6 +965,9 @@
 [submodule "vendor/grammars/vscode-lean"]
 	path = vendor/grammars/vscode-lean
 	url = https://github.com/leanprover/vscode-lean
+[submodule "vendor/grammars/vscode-prisma"]
+	path = vendor/grammars/vscode-prisma
+	url = https://github.com/prisma/vscode-prisma
```
Comment (pchaigno) [inline .gitmodules]: This project is missing a license. We can't use it without a license (it is currently not open source).

File: samples/Prisma/datasource-mongo.prisma [hunk 3]
```diff
@@ -0,0 +1,41 @@
+datasource mongo {
```
Comment (pchaigno) [inline samples/Prisma/datasource-mongo.prisma]: We use these sample files to train a Bayesian classifier. I'm worried that the 5 files you've added are more similar than usual files would be (all from similar database context). The classifier could for instance learn that `createdAt` is a good indication of a Prisma file, when that is likely not the case. Any chance you could change them to more representative sample files?

File: lib/linguist/languages.yml (hunk 1 above)
Comment (rajinwonderland) [reply lib/linguist/languages.yml]: Hey @pchaigno, just asked the guys at prisma to go ahead and add the missing license to the grammar repo. Should be up soon!

I believe that's my [prettier](https://github.com/prettier/prettier) settings on vscode that are auto editing those files. I can go ahead and clean them up if you'd like and disable it from now on!

File: .gitmodules (hunk 2 above)
Comment (rajinwonderland) [reply .gitmodules]: Hey @pchaigno, appreciate the help! 

Just asked the guys at prisma to go ahead and add the missing license to the grammar repo. Should be up soon! I'll push an update as soon as they have that finished!

File: samples/Prisma/datasource-mongo.prisma (hunk 3 above)
Comment (rajinwonderland) [reply samples/Prisma/datasource-mongo.prisma]: Hey @pchaigno, sure I can definitely differentiate the schema examples!

**Comment by rajinwonderland** [comment]
@pchaigno regarding the `config.prisma` files, I'm honestly not sure what or where they come from, however, they are not related to this pull request. Any ideas on how we can make sure we differentiate these files?

File: lib/linguist/languages.yml [hunk 4]
```diff
@@ -5969,4 +5976,4 @@ xBase:
   - ".prw"
   tm_scope: source.harbour
   ace_mode: text
-  language_id: 421
+  language_id: 421
```
Comment (pchaigno) [inline lib/linguist/languages.yml]: Could you please add the newline back?

File: vendor/licenses/grammar/vscode-prisma.txt [hunk 5]
```diff
@@ -0,0 +1,6 @@
+---
+type: grammar
+name: vscode-prisma
+version: ef6481235feb87fb7c0888745bbbb6ed6861175e
+license: apache-2.0
+---
```
Comment (pchaigno) [inline vendor/licenses/grammar/vscode-prisma.txt]: The license text appears to be missing here. Did you use the add-grammar script?

File: samples/Prisma/datasource-mongo.prisma (hunk 3 above)
Comment (pchaigno) [reply samples/Prisma/datasource-mongo.prisma]: Thanks! The samples look better now :+1:

**Comment by pchaigno** [comment]
> @pchaigno regarding the `config.prisma` files, I'm honestly not sure what or where they come from, however, they are not related to this pull request. Any ideas on how we can make sure we differentiate these files?

There are only 116 such files, so we can probably ignore them for now. Since Prisma is a data language, having these files detected as Prisma shouldn't have too many user-visible impacts.

File: vendor/licenses/grammar/vscode-prisma.txt (hunk 5 above)
Comment (schickling) [reply vendor/licenses/grammar/vscode-prisma.txt]: Just adjusted it.

File: lib/linguist/languages.yml (hunk 4 above)
Comment (rajinwonderland) [reply lib/linguist/languages.yml]: sure

**Comment by rajinwonderland** [comment]
Hey @pchaingo still having trouble adding the license from the vscode prisma repo. Any help here would be greatly appreciated!

**Comment by pchaigno** [comment]
Did you use `scripts/add-grammar` to add that grammar?

**Comment by rajinwonderland** [comment]
Yes!

File: vendor/licenses/grammar/vscode-prisma.txt (hunk 5 above)
Comment (lildude) [reply vendor/licenses/grammar/vscode-prisma.txt]: 🤔 the actual license text should have been pulled in when running `add-grammar`. I've just tested locally and it works for me:

```console
$ script/add-grammar https://github.com/prisma/vscode-prisma 
Checking docker is installed and running
$ docker ps
Registering new submodule: vendor/grammars/vscode-prisma
$ git submodule add -f https://github.com/prisma/vscode-prisma vendor/grammars/vscode-prisma
$ script/grammar-compiler add vendor/grammars/vscode-prisma
Confirming license
$ script/licensed --module /Users/lildude/github/linguist/vendor/grammars/vscode-prisma
Updating grammar documentation in vendor/README.md
$ bundle exec rake samples
$ script/sort-submodules
$ script/list-grammars
$ head vendor/licenses/grammar/vscode-prisma.txt
---
type: grammar
name: vscode-prisma
version: e90368f8d15ed30fe95f3f83e252132a3091f845
license: apache-2.0
---
                                 Apache License
                           Version 2.0, January 2004
                        http://www.apache.org/licenses/
$
```
Can you please try again or manually add the license text to this file.

**Pull request closed without merging**